editor/scene_tree_dock.cpp

```diff
@@ -2580,6 +2590,44 @@ void SceneTreeDock::_create() {
 
 		// This works because editor_selection was cleared and populated with last created node in _do_create()
 		Node *last_created = editor_selection->get_selected_node_list().front()->get();
+
+		if (center_parent) {
+			// Find parent type and only average positions of relevant nodes
+			Node3D *parent_node_3d = Object::cast_to<Node3D>(last_created);
+			if (parent_node_3d) {
+				Vector3 position;
+				uint32_t node_count = 0;
+				for (Node *node : nodes) {
```

Comment:
```suggestion
				for (const Node *node : nodes) {
```
(if it's possible; you need to make the Node3D const too)